File: packages/engine/src/cpus/6502/instructions/sed.js
import { CycleType } from '..'

/**
 * Set Decimal Mode
 *
 * @example
 * MODE          SYNTAX        HEX LEN TIM
 * Implied       SED           $F8  1   2
 */
export function* sed() {
  this.decimalMode = true
  yield CycleType.GET
}

File: packages/engine/src/cores/famibob/cpu/instructions/sed.zig
const std = @import("std");
const CPU = @import("../cpu.zig").CPU;
const Addressing = @import("../opcode.zig").Addressing;

// Set Decimal Mode
//
// MODE          SYNTAX        HEX LEN TIM
// Implied       SED           $F8  1   2

pub fn sed(self: *CPU, addressing: Addressing) void {
    _ = addressing;
    self.d = true;
}
